sycl/include/syclcompat/launch.hpp

```diff
@@ -118,6 +118,22 @@ launch(const dim3 &grid, const dim3 &threads, Args... args) {
   return launch<F>(grid, threads, get_default_queue(), args...);
 }
 
+// Overload taking zero-argument function object
+template <class F, int Dim>
+std::enable_if_t<std::is_invocable_v<const F &>, sycl::event>
+launch(const F &f, const sycl::nd_range<Dim> &range,
+       sycl::queue q = get_default_queue()) {
+  return q.parallel_for(detail::transform_nd_range<Dim>(range),
+                        [=](sycl::nd_item<Dim>) { f(); });
+}
+// Alternative launch through dim3 objects
+template <class F>
+std::enable_if_t<std::is_invocable_v<const F &>, sycl::event>
```

Comment:
I know that this is unchanged, but I think you will want an overload of these launch functions that do not return an event, and that use the new enqueue_functions API (nd_launch): see https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_functions.asciidoc 
This is much closer to native cuda and does lead to a very noticeable performance improvement for small kernels (relevant to cutlass), because the overhead of creating events in cuda/hip is big.

I do not know whether having the function calling nd_launch in the implementation might change other design decisions here.

Reply:
I agree we want a version which doesn't return an event and calls nd_launch. I don't think internally calling nd_launch will change the design decision here.

An overload isn't a good way to add a version which doesn't return an event. Given that one can't overload by return type, it would need to be a property or something like it to opt-out of the event. But getting an event back should be opt-in nor opt-out.

It would be best if we could just remove the return value. But I assume we don't want to break API.

If so I think we need to add a new function like `launch2` or `launch_fast` or something similar which doesn't return an event.